### PR TITLE
Use Webserver

### PR DESCRIPTION
**Pull request by krishnans2006**
Compatibility with https://github.com/tj-uav/FlightSoftware/pull/11 to use a new HTTP-based image transfer system, plus more fixes.

This is a preliminary PR before merging into https://github.com/tj-uav/GroundStation/pull/73.

Warning: May cause conflicts with uncommitted changes from @pizzalemon, although I would recommend merging this into `uavhandler` first and then resolving the (hopefully very few) conflicts with that code later.

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -34,9 +34,9 @@ const App = () => {
 						<Submissions />
 					</Route>
 
-					<Route path="/antenna-tracker">
-						<AntennaTracker />
-					</Route>
+					{/*<Route path="/antenna-tracker">*/}
+					{/*	<AntennaTracker />*/}
+					{/*</Route>*/}
```
Comment (pizzalemon) [inline client/src/App.js]: Why commit this though.

File: client/src/components/FlightMap.js [hunk 2]
```diff
@@ -195,7 +195,8 @@ const FlightPlanMap = props => {
 			>
 				<TileLayer
 					attribution='&amp;copy <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
-					url={online ? "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" : "/map/{z}/{x}/{y}.png"}
+					// url={online ? "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" : "/map/{z}/{x}/{y}.png"}
+					url={"https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"}
```
Comment (pizzalemon) [inline client/src/components/FlightMap.js]: This change shouldn't be commited.

File: client/src/pages/FlightData/tabs/Logs.js [hunk 3]
```diff
@@ -86,6 +87,7 @@ const Logs = () => {
 	return (
 		<StyledContainer>
 			<CheckboxList name="logFilter" onChange={(e) => {
+					console.log(e.target.value)
```
Comment (pizzalemon) [inline client/src/pages/FlightData/tabs/Logs.js:90]: This shouldn't be commited.

File: .gitignore [hunk 4]
```diff
@@ -152,6 +152,7 @@ cython_debug/
 !/server/assets/odlc_images/sample.png
 /server/handlers/uav/uav_mission.txt
 /server/handlers/ugv/ugv_mission.txt
+/client/public/map/*/*/*
```
Comment (pizzalemon) [inline .gitignore:155]: Very relevant to commit.

File: server/groundstation.py [hunk 5]
```diff
@@ -92,14 +94,26 @@ def ugv_thread(self):
             time.sleep(0.1)
 
     def image_thread(self):
-        if not self.config["uav"]["images"]["type"] == "prod":  # Initialize a socket connection
-            self.image.socket_connect()
+        if self.config["uav"]["images"]["type"] == "prod":  # Initialize a socket connection
```
Comment (pizzalemon) [inline server/groundstation.py:97]: Socket connection??? Thought we were getting rid of that.

File: server/groundstation.py [hunk 6]
```diff
@@ -92,14 +94,26 @@ def ugv_thread(self):
...
+            while True:
+                time.sleep(1)
```
Comment (pizzalemon) [inline server/groundstation.py:99]: Bruh moment.

File: server/handlers/image.py [hunk 7]
```diff
@@ -67,14 +52,14 @@ def retreive_images(self):
         return False
 
     def process_image(self, image):
-        if random() < 0.05:  # 5% chance that the image is of an ODLC
+        if random() < 1:  # 5% chance that the image is of an ODLC
```
Comment (pizzalemon) [inline server/handlers/image.py:56]: Why the change here

File: server/handlers/image.py [hunk 8]
```diff
@@ -67,14 +52,14 @@ def retreive_images(self):
...
-            self.gs.interop.odlc.add_to_queue(
+            self.gs.interop.odlc_add_to_queue(
                 image,
                 "standard",
-                random() * 90,
-                random() * -90,
-                randint(0, 360),
+                self.gs.uav.lat,
+                self.gs.uav.lon,
+                self.gs.uav.orientation["yaw"],
```
Comment (pizzalemon) [inline server/handlers/image.py:63]: Very relevant to Flight Software server. 

You might want to split up this PR into two commits or something.

File: server/handlers/ugv/dummy.py [hunk 9]
```diff
@@ -69,7 +74,7 @@ def readmission(filename):
 class DummyUGVHandler:
     def __init__(self, gs, config):
         self.logger = logging.getLogger("groundstation")
-        self.gs = gs
+        self.gs: GroundStation = gs
```
Comment (pizzalemon) [inline server/handlers/ugv/dummy.py:77]: I don't really care that much, but if you want you could put all these type annotations and small other stuff into a separate code cleanup commit.

File: server/handlers/ugv/dummy.py [hunk 10]
```diff
@@ -116,8 +121,12 @@ def update(self):
...
                 self.droppos = self.gs.interop.get_data("ugv")
                 self.droppos = self.droppos["result"]
-            self.lat = self.droppos["drop"]["latitude"] + (random.random() - 0.5) / 2000
-            self.lon = self.droppos["drop"]["longitude"] + (random.random() - 0.5) / 2000
+            if not self.lat:
+                self.lat = self.droppos["drop"]["latitude"]# + (random.random() - 0.5) / 2000
+                self.lon = self.droppos["drop"]["longitude"]# + (random.random() - 0.5) / 2000
+            self.lat += (random.random() - 0.5) / 80000
+            self.lon += (random.random() - 0.5) / 80000
+            # ^^ remove after frr
```
Comment (pizzalemon) [inline server/handlers/ugv/dummy.py:129]: ^^ remove after frr

for real

File: client/src/App.js (hunk 1 above)
Comment (krishnans2006) [reply client/src/App.js]: Cause Antenna Tracker is irrelevant

**Review by pizzalemon — APPROVED** [review]
Alright, fine.
